Code/Editor/Lib/Tests/test_AzAssetBrowserRequestHandler.cpp

```diff
@@ -0,0 +1,678 @@
+
+        DragAndDropEventsBus::Event(EditorViewport, &DragAndDropEvents::Drop, &dropEvent, ctx);
+
+        mockHandler.BusDisconnect();
+
+        EXPECT_EQ(dropEvent.isAccepted(), true);
+        ASSERT_EQ(m_createdEntities.size(), 1);
+
+        for (auto entity : m_createdEntities)
+        {
+           MockEditorComponent* mockComponent = entity->FindComponent<MockEditorComponent>();
+           ASSERT_NE(mockComponent, nullptr);
+
+           // we expect the 'aaaa' asset, not the zzzz one.
+           EXPECT_EQ(mockComponent->m_primaryAssetSet, AZ::Data::AssetId(m_uuidOfSource1, 2))
+               << "Invalid component spawned.  Should have spawned the aaa one alphabetically.";
+        }
+    }
+
+    // this is the same test as above, but this time, picks the second source file
+    // the one with 2 different types of assets.
+    // it should pick the second asset, despite it being later alphabetically, due to the priority override bus.
+    TEST_F(AzAssetBrowserRequestHandlerFixture, DragEnterTest_DragSource_WithDefaultBehavior_PrioritySystemWorks)
+    {
+        using namespace AzToolsFramework;
+        using namespace AzToolsFramework::AssetBrowser;
+        using namespace AzQtComponents;
+        using namespace AzQtComponents::DragAndDropContexts;
+
+        AzAssetBrowserRequestHandler browser;
+        AzQtComponents::ViewportDragContext ctx;
+        ctx.m_hitLocation = AZ::Vector3::CreateZero();
+
+        // simulate dragging and dropping the source file:
+        QMimeData mimeData;
+        AZStd::vector<const AssetBrowserEntry*> entries;
+        // use the 2nd source file - the one that has an actor and a model as its children
+        entries.push_back(SourceAssetBrowserEntry::GetSourceByUuid(m_uuidOfSource2));
+        Utils::ToMimeData(&mimeData, entries);
+
+        QDropEvent dropEvent(
+            QPoint(0, 0), Qt::DropAction::CopyAction, &mimeData, Qt::MouseButton::LeftButton, Qt::KeyboardModifier::NoModifier);
+
+        // when asset of type m_assetTypeOfModel is dragged, spawn a s_MockEditorComponentTypeId.
+        // when asset of type m_assetTypeOfActor is dragged, spawn a s_MockEditorComponentTypeId also - but high priority
+        testing::NiceMock<AssetTypeInfoBusHandlerMock> mockHandler1;
+        testing::NiceMock<AssetTypeInfoBusHandlerMock> mockHandler2;
+
+        // the actor handler should have a higher priority, so it should always 'win'
+        mockHandler1.InstallHandlersFor(AZ::Uuid(MockEditorComponent::s_MockEditorComponentTypeId), m_assetTypeOfModel);
+        mockHandler2.InstallHandlersFor(AZ::Uuid(MockEditorComponent::s_MockEditorComponentTypeId), m_assetTypeOfActor, 10);
+
+        DragAndDropEventsBus::Event(EditorViewport, &DragAndDropEvents::Drop, &dropEvent, ctx);
+
+        mockHandler1.BusDisconnect();
+        mockHandler2.BusDisconnect();
+
+        EXPECT_EQ(dropEvent.isAccepted(), true);
+        ASSERT_EQ(m_createdEntities.size(), 1);
```

Comment:
When a failure occurs in googletest, it outputs the first parameter as the "expected" value and the second parameter as the actual.

Also for function that return a boolean, `EXPECT_TRUE` or `EXPECT_FALSE` should be used